evals/elsuite/basic/includes.py

```diff
@@ -13,24 +13,31 @@ def __init__(
         samples_jsonl: str,
         *args,
         max_tokens: int = 500,
+        completion_fn: utils.CompletionFn = evals.completion_query,
         **kwargs,
     ):
         super().__init__(model_specs, *args, **kwargs)
         self.max_tokens = max_tokens
         self.samples_jsonl = samples_jsonl
+        self._completion_fn = completion_fn
 
     def eval_sample(self, sample: Any, *_):
-        sampled = evals.sample_freeform(
-            self.model_spec, sample["input"], max_tokens=self.max_tokens
+        response, actual_prompt, metadata = self._completion_fn(
+            prompt=sample["input"],
+            max_tokens=self.max_tokens,
+            model_spec=self.model_spec,
         )
+        sampled: str = evals.postprocess_sample_freeform(
```

Comment:
If we made `sample_freeform` support arbitrary `_completion_fn`, do we still need these broken out in this way for each eval?